geoguide/client/static/src/pageEnvironment/index.js

```diff
@@ -38,6 +38,60 @@ let sizeModifierElement = document.querySelector('#sizeModifier')
 let sizeModifier = sizeModifierElement.value
 let sizeModifierMax = {}
 
+if (typeof(Storage) !== "undefined") {
+  let ls;
+  if(localStorage.getItem(JSON.parse(document.querySelector('#dataset_json').innerHTML).filename) == null) {
+    ls = {
+      "colorModifier": document.querySelector('#colorModifier').selectedIndex,
+      "sizeModifier": document.querySelector('#sizeModifier').selectedIndex,
+      "timelimit": document.getElementById("timelimit").value,
+      "sigma": document.getElementById("sigma").value,
+      "kvalue": document.getElementById("kvalue").value,
+      "onlyfilteredpoints": $("#onlyfilteredpoints").prop(':checked')
+    };
+    localStorage.setItem(JSON.parse(document.querySelector('#dataset_json').innerHTML).filename, JSON.stringify(ls));
+  } else {
+     ls = JSON.parse(localStorage.getItem(JSON.parse(document.querySelector('#dataset_json').innerHTML).filename));
```

Comment:
Why not save in a variable the value of `JSON.parse(document.querySelector('#dataset_json').innerHTML).filename`? You see you are parsing twice at least...